### ISSUE TITLE
hard track not visible in gazebo

### ISSUE THREAD

**Issue by mattcamp** [aws-robotics/aws-robomaker-sample-application-deepracer#11]
I'm having a problem that whenever I set ```WORLD_NAME=hard_track``` the track does not appear in gazebo.

The car appears but nothing else.

easy_track and medium_track work fine.

This is a totally fresh install on ros melodic on Ubuntu 18.04.

```
Checking log directory for disk usage. This may take awhile.
Press Ctrl-C to interrupt
Done checking log file disk usage. Usage is <1GB.

started roslaunch server http://matt-virtual-machine:34543/

SUMMARY
========

PARAMETERS
 * /JOB_TYPE: TRAINING
 * /MODEL_S3_BUCKET: sagemaker-eu-west...
 * /MODEL_S3_PREFIX: deepracer-matt-vm...
 * /ROS_AWS_REGION: eu-west-1
 * /WORLD_NAME: hard_track
 * /gazebo/enable_ros_network: True
 * /racecar/joint_state_controller/publish_rate: 50
 * /racecar/joint_state_controller/type: joint_state_contr...
 * /racecar/left_front_wheel_velocity_controller/joint: left_front_wheel_...
 * /racecar/left_front_wheel_velocity_controller/pid/d: 0.0
 * /racecar/left_front_wheel_velocity_controller/pid/i: 0.0
 * /racecar/left_front_wheel_velocity_controller/pid/i_clamp: 0.0
 * /racecar/left_front_wheel_velocity_controller/pid/p: 0.5
 * /racecar/left_front_wheel_velocity_controller/type: effort_controller...
 * /racecar/left_rear_wheel_velocity_controller/joint: left_rear_wheel_j...
 * /racecar/left_rear_wheel_velocity_controller/pid/d: 0.0
 * /racecar/left_rear_wheel_velocity_controller/pid/i: 0.0
 * /racecar/left_rear_wheel_velocity_controller/pid/i_clamp: 0.0
 * /racecar/left_rear_wheel_velocity_controller/pid/p: 1.0
 * /racecar/left_rear_wheel_velocity_controller/type: effort_controller...
 * /racecar/left_steering_hinge_position_controller/joint: left_steering_hin...
 * /racecar/left_steering_hinge_position_controller/pid/d: 0.5
 * /racecar/left_steering_hinge_position_controller/pid/i: 0.0
 * /racecar/left_steering_hinge_position_controller/pid/p: 1.0
 * /racecar/left_steering_hinge_position_controller/type: effort_controller...
 * /racecar/right_front_wheel_velocity_controller/joint: right_front_wheel...
 * /racecar/right_front_wheel_velocity_controller/pid/d: 0.0
 * /racecar/right_front_wheel_velocity_controller/pid/i: 0.0
 * /racecar/right_front_wheel_velocity_controller/pid/i_clamp: 0.0
 * /racecar/right_front_wheel_velocity_controller/pid/p: 0.5
 * /racecar/right_front_wheel_velocity_controller/type: effort_controller...
 * /racecar/right_rear_wheel_velocity_controller/joint: right_rear_wheel_...
 * /racecar/right_rear_wheel_velocity_controller/pid/d: 0.0
 * /racecar/right_rear_wheel_velocity_controller/pid/i: 0.0
 * /racecar/right_rear_wheel_velocity_controller/pid/i_clamp: 0.0
 * /racecar/right_rear_wheel_velocity_controller/pid/p: 1.0
 * /racecar/right_rear_wheel_velocity_controller/type: effort_controller...
 * /racecar/right_steering_hinge_position_controller/joint: right_steering_hi...
 * /racecar/right_steering_hinge_position_controller/pid/d: 0.5
 * /racecar/right_steering_hinge_position_controller/pid/i: 0.0
 * /racecar/right_steering_hinge_position_controller/pid/p: 1.0
 * /racecar/right_steering_hinge_position_controller/type: effort_controller...
 * /robot_description: <?xml version="1....
 * /rosdistro: melodic
 * /rosversion: 1.14.3
 * /use_sim_time: True

NODES
  /racecar/
    controller_manager (controller_manager/spawner)
  /
    agent (deepracer_simulation/run_local_rl_agent.sh)
    better_odom (topic_tools/relay)
    gazebo (gazebo_ros/gzserver)
    gazebo_gui (gazebo_ros/gzclient)
    racecar_spawn (gazebo_ros/spawn_model)
    robot_state_publisher (robot_state_publisher/robot_state_publisher)
    servo_commands (deepracer_simulation/servo_commands.py)

auto-starting new master
process[master]: started with pid [92044]
ROS_MASTER_URI=http://localhost:11311

setting /run_id to fcc120f8-2fa9-11e9-ba3d-000c2988e0bf
process[rosout-1]: started with pid [92055]
started core service [/rosout]
process[gazebo-2]: started with pid [92062]
process[gazebo_gui-3]: started with pid [92067]
process[racecar_spawn-4]: started with pid [92072]
process[racecar/controller_manager-5]: started with pid [92073]
process[robot_state_publisher-6]: started with pid [92074]
process[servo_commands-7]: started with pid [92075]
process[better_odom-8]: started with pid [92080]
+ export NODE_TYPE=SIMULATION_WORKER
+ NODE_TYPE=SIMULATION_WORKER
+ python3 -m markov.single_machine_training_worker
process[agent-9]: started with pid [92086]
[INFO] [1550074276.622891, 0.000000]: Controller Spawner: Waiting for service controller_manager/load_controller
[ INFO] [1550074276.878730333]: Finished loading Gazebo ROS API Plugin.
[ INFO] [1550074276.880908370]: waitForService: Service [/gazebo/set_physics_properties] has not been advertised, waiting...
[ INFO] [1550074276.911774212]: waitForService: Service [/gazebo_gui/set_physics_properties] has not been advertised, waiting...
[ INFO] [1550074276.916299451]: Finished loading Gazebo ROS API Plugin.
[INFO] [1550074277.470849, 0.000000]: Loading model XML from ros parameter robot_description
[INFO] [1550074277.528489, 0.000000]: Waiting for service /gazebo/spawn_urdf_model
[ INFO] [1550074277.927572187, 0.025000000]: waitForService: Service [/gazebo/set_physics_properties] is now available.
[ INFO] [1550074278.041847933, 0.154000000]: Physics dynamic reconfigure ready.
[INFO] [1550074278.137237, 0.389000]: Calling service /gazebo/spawn_urdf_model
[INFO] [1550074278.410802, 0.563000]: Spawn status: SpawnModel: Successfully spawned entity
[racecar_spawn-4] process has finished cleanly
log file: /home/matt/.ros/log/fcc120f8-2fa9-11e9-ba3d-000c2988e0bf/racecar_spawn-4*.log
[ INFO] [1550074278.961235120, 0.563000000]: Camera Plugin: Using the 'robotNamespace' param: '/'
[ INFO] [1550074278.990773694, 0.563000000]: Camera Plugin (ns = /)  <tf_prefix_>, set to ""
[ INFO] [1550074279.217262002, 0.563000000]: Loading gazebo_ros_control plugin
[ INFO] [1550074279.217471212, 0.563000000]: Starting gazebo_ros_control plugin in namespace: /racecar
[ INFO] [1550074279.221520789, 0.563000000]: gazebo_ros_control plugin is waiting for model URDF in parameter [/robot_description] on the ROS param server.
Using custom preset file from Markov presets directory!
Creating graph - name: BasicRLGraphManager
[ WARN] [1550074279.347765025, 0.563000000]: Deprecated syntax, please prepend 'hardware_interface/' to 'EffortJointInterface' within the <hardwareInterface> tag in joint 'left_rear_wheel_joint'.
[ WARN] [1550074279.349632622, 0.563000000]: Deprecated syntax, please prepend 'hardware_interface/' to 'EffortJointInterface' within the <hardwareInterface> tag in joint 'right_rear_wheel_joint'.
[ WARN] [1550074279.350340224, 0.563000000]: Deprecated syntax, please prepend 'hardware_interface/' to 'EffortJointInterface' within the <hardwareInterface> tag in joint 'left_steering_hinge_joint'.
[ WARN] [1550074279.351292147, 0.563000000]: Deprecated syntax, please prepend 'hardware_interface/' to 'EffortJointInterface' within the <hardwareInterface> tag in joint 'right_steering_hinge_joint'.
[ WARN] [1550074279.352570990, 0.563000000]: Deprecated syntax, please prepend 'hardware_interface/' to 'EffortJointInterface' within the <hardwareInterface> tag in joint 'left_front_wheel_joint'.
[ WARN] [1550074279.353282304, 0.563000000]: Deprecated syntax, please prepend 'hardware_interface/' to 'EffortJointInterface' within the <hardwareInterface> tag in joint 'right_front_wheel_joint'.
[ INFO] [1550074279.365956851, 0.563000000]: Loaded gazebo_ros_control.
******* World name is: hard_track *********
[INFO] [1550074279.639861, 0.768000]: Controller Spawner: Waiting for service controller_manager/switch_controller
[INFO] [1550074279.643279, 0.768000]: Controller Spawner: Waiting for service controller_manager/unload_controller
[INFO] [1550074279.646346, 0.768000]: Loading controller: left_rear_wheel_velocity_controller
[INFO] [1550074279.742422, 0.809000]: Loading controller: right_rear_wheel_velocity_controller
[INFO] [1550074279.809821, 0.819000]: Loading controller: left_front_wheel_velocity_controller
[INFO] [1550074279.871939, 0.831000]: Loading controller: right_front_wheel_velocity_controller
[INFO] [1550074279.928403, 0.849000]: Loading controller: left_steering_hinge_position_controller
Total Reward Reward=0.00 Total Steps=0.00
[INFO] [1550074280.088112, 0.917000]: Loading controller: right_steering_hinge_position_controller
[INFO] [1550074280.164898, 0.946000]: Loading controller: joint_state_controller
[INFO] [1550074280.186326, 0.947000]: Controller Spawner: Loaded controllers: left_rear_wheel_velocity_controller, right_rear_wheel_velocity_controller, left_front_wheel_velocity_controller, right_front_wheel_velocity_controller, left_steering_hinge_position_controller, right_steering_hinge_position_controller, joint_state_controller
[INFO] [1550074280.192177, 0.950000]: Started controllers: left_rear_wheel_velocity_controller, right_rear_wheel_velocity_controller, left_front_wheel_velocity_controller, right_front_wheel_velocity_controller, left_steering_hinge_position_controller, right_steering_hinge_position_controller, joint_state_controller
Step No=0.00 Step Reward=1.00
Step No=1.00 Step Reward=1.00
Creating agent - name: agent
2019-02-13 16:11:36.939611: I tensorflow/core/platform/cpu_feature_guard.cc:141] Your CPU supports instructions that this TensorFlow binary was not compiled to use: AVX2 FMA
^C[agent-9] killing on exit

```

**Comment by x77a1** [comment]
Are you using Gazebo 9?

**Comment by mattcamp** [comment]
Yes.

I did a standard full desktop install of ROS melodic which installed Gazebo 9.0.0

**Comment by x77a1** [comment]
Hi Matt,

Thank you for bringing up this issue. We currently only support ROS Kinetic and Gazebo 7 on Ubuntu Xenial. We are working on supporting ROS Melodic and Gazebo 9 but its not yet supported. As part of that initiative we will try to fix this issue. However till we have support for Gazebo 9 could you please try to run it using Gazebo 7 and ROS Kinetic on Ubuntu Xenial? 

P.S I will leave this issue open till we have fix for this issue.

**Comment by exitNA** [comment]
i have the same problem, I use kinetic with gazebo9. I can see the car but the tack not show.
![qq 20190216221620](https://user-images.githubusercontent.com/5160007/52901018-97550680-3238-11e9-92ca-3397a9311784.png)

**Comment by cgibb** [comment]
Closing this, as it is fixed by PR #14. Please reopen if the fix is not working for either of you.
